tests/cpp17/test_cpp17.cpp

```diff
@@ -36,8 +36,31 @@ namespace cpp11 {
 #include "../monster_test_generated.h"
 }  // namespace cpp11
 
+void CreateTableByTypeTest() {
+  flatbuffers::FlatBufferBuilder builder;
+
+  // We will create an object of this type using only the type.
+  using type_to_create_t = cpp17::MyGame::Example::Stat;
+
+  {
+    auto table = type_to_create_t::Traits::Create(
+        builder, builder.CreateString("my_id"), 42, 7);
+    // Be sure that the correct return type was inferred.
+    static_assert(
+        std::is_same_v<decltype(table), flatbuffers::Offset<type_to_create_t>>);
+    builder.Finish(table);
+  }
+
```

Comment:
This is a comment, without changes request.
In C++17 we can define in-place lambda and invoke it immediately.
For example:
```cpp
  [&builder]() {
    auto id = builder.CreateString("my_id");
    auto table = type_to_create_t::Traits::Create(builder, id, 42, 7);
  }(); // <- invoke
```
or return value:
```cpp
  auto table = [&builder]() {
    auto id = builder.CreateString("my_id");
    return type_to_create_t::Traits::Create(builder, id, 42, 7);
  }();
```
This gives control over the visibility of previously declared variables and prevents mistakes.
This is not actual for the small test function however it can be useful.

Reply:
Done.

Reply:
I didn't know that expression `[&builder]{ ... }()` can be used for lambda without arguments.
Thanks.